pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -147,8 +163,35 @@ func (c *Controller) logPacket(pktIn *ofctrl.PacketIn) error {
 		return fmt.Errorf("received error while handling packetin for NetworkPolicy: %v", err)
 	}
 
-	// Store log file
-	AntreaPolicyLogger.Printf("%s %s %s %s SRC: %s DEST: %s %d %s", ob.tableName, ob.npRef, ob.disposition, ob.ofPriority, ob.srcIP, ob.destIP, ob.pktLength, ob.protocolStr)
+	// Create 1 sec buffers for deduplicating non-Allow packet log.
+	logMsg := fmt.Sprintf("%s %s %s %s SRC: %s DEST: %s %d %s", ob.tableName, ob.npRef, ob.disposition, ob.ofPriority, ob.srcIP, ob.destIP, ob.pktLength, ob.protocolStr)
+	if ob.disposition == openflow.DispositionToString[openflow.DispositionAllow] {
+		AntreaPolicyLogger.Printf(logMsg)
+	} else {
+		logDeduplication.logMutex.Lock()
+		defer logDeduplication.logMutex.Unlock()
+		_, ok := logDeduplication.logRecord[logMsg]
+		// Increase count if duplicated within 1 sec, create buffer otherwise.
+		if ok {
+			logDeduplication.logRecord[logMsg].count++
```

Comment:
could also consider separate methods for logSafeGuard to CRUD the counter in thread safe manner. you can see example for ipAllocationMap in egress/controller.go